### PR TITLE
*: fix compile fail for test

### LINKED ISSUE #17707: *: Some tests fail to compile
### What version of TiKV are you using?

Master

Some test (`test_load_region_became_leader` and tests under module without enable failpoint but import raftstore) fail to compile.

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #17707

What's Changed:

```commit-message
fix compile fail for test
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/hybrid_engine/Cargo.toml [hunk 1]
```diff
@@ -39,3 +39,4 @@ keys = { workspace = true }
 [dev-dependencies]
 tempfile = "3.0"
 test_util = { workspace = true }
+raftstore = { workspace = true, features = ["testexport"] }
```
Comment (SpadeA-Tang) [inline components/hybrid_engine/Cargo.toml:42]: For test `test_load_region_became_leader`.

File: components/raftstore/src/store/fsm/peer.rs [hunk 2]
```diff
@@ -2663,21 +2663,25 @@ where
                 && !is_initialized_peer
                 && msg_type == target_msg_type
         };
+        #[cfg(feature = "failpoints")]
```
Comment (SpadeA-Tang) [inline components/raftstore/src/store/fsm/peer.rs:2666]: fp_enable is cfg under failpoints, so the the code use it should cfg under failpoints

**Comment by overvenus** [comment]
What's your compile command?

**Comment by SpadeA-Tang** [comment]
> What's your compile command?

cargo test --package hybrid_engine --lib -- observer::load_eviction::tests::test_load_region_became_leader --exact --nocapture

**Comment by SpadeA-Tang** [comment]
> > What's your compile command?
> 
> cargo test --package hybrid_engine --lib -- observer::load_eviction::tests::test_load_region_became_leader --exact --nocapture

triggered by the vscode `Run Test`

File: components/raftstore/src/store/fsm/peer.rs (hunk 2 above)
Comment (glorv) [reply components/raftstore/src/store/fsm/peer.rs:2666]: `fail_point` macro expand to noop if `failpoints` is disabled, so there seems no need to add this explicit cfg attr, do you meet a compiling error here?

Comment (SpadeA-Tang) [reply components/raftstore/src/store/fsm/peer.rs:2666]: yes, you may run tests in in_memory_engine/tests/failpoints
such as `cargo test --package in_memory_engine --test failpoints --features failpoints -- test_memory_engine::test_gc_worker --exact --nocapture`

Comment (glorv) [reply components/raftstore/src/store/fsm/peer.rs:2666]: So this is a problem of in_memory_engine rather that raftstore, you should enable `"raftstore/failpoints"` in in_memory_engine's failpoints feature(in in_memory_engine/Cargo.toml)

**Review by glorv — APPROVED** [review]

**Comment by overvenus** [comment]
Is this PR still relevant? I saw similar changes in #17712.

**Comment by SpadeA-Tang** [comment]
> Is this PR still relevant? I saw similar changes in #17712.

Yes. I reverted the change in #17712 and also, the changes made in it is not enough.

**Review by overvenus — APPROVED** [review]

**Pull request merged**